packages/common/src/test-utils.ts

```diff
@@ -0,0 +1,170 @@
+import crypto from 'crypto';
+import path from 'path';
+import RandExp from 'randexp';
+import fs from 'fs-extra';
+import { ExecutionContext } from 'ava';
+
+export { readJsonFile as readJsonFixture } from './FileUtils';
+
+export const inTestMode = () => process.env.NODE_ENV === 'test';
+
+/**
+ * Helper function to throw error for unit test exports
+ * @throws {Error}
+ */
+export const throwTestError = () => {
+  throw new Error('This function is only exportable when NODE_ENV === test for unit test purposes');
+};
+
+/**
+ * Generate a [40 character] random string
+ *
+ * @param {number} numBytes - number of bytes to use in creating a random string
+ *                 defaults to 20 to produce a 40 character string
+ * @returns {string} - a random string
+ */
+export const randomString = (numBytes = 20) =>
+  crypto.randomBytes(numBytes).toString('hex');
+
+/**
+ * Postpend a [10-character] random string to input identifier.
+ *
+ * @param {string} id - identifer to return
+ * @param {number} numBytes - number of bytes to use to compute random
+ *                 extension. Default 5 to produce 10 characters..
+ * @returns {string} - a random string
+ */
+export const randomId = (id: string, numBytes = 5) =>
+  `${id}${exports.randomString(numBytes)}`;
+
+/**
+ * Generate a random for the given scale.
+ *
+ * Defaults to a number between 1 and 10.
+ *
+ * @param {number} scale - scale for the random number. Defaults to 10.
+ * @returns {number} - a random number
+ */
+export const randomNumber = (scale = 10) => Math.ceil(Math.random() * scale);
+
+/**
+ * Create a random granule id from the regular expression
+ *
+ * @param {string} regex - regular expression string
+ * @returns {string} - random granule id
+ */
+export const randomStringFromRegex = (regex: string) =>
+  new RandExp(regex).gen();
+
+/**
+ * Validate an object using json-schema
+ *
+ * Issues a test failure if there were validation errors
+ *
+ * @param {Object} t - an ava test
+ * @param {string} schemaFilename - the filename of the schema
+ * @param {Object} data - the object to be validated
+ * @returns {Promise<undefined>}
+ */
+async function validateJSON(
+  t: ExecutionContext,
+  schemaFilename: string,
+  data: unknown
+) {
+  const schemaName = path.basename(schemaFilename).split('.')[0];
+  const schema = await fs.readFile(schemaFilename, 'utf8').then(JSON.parse);
+  const ajv = new Ajv();
+  const valid = <boolean>ajv.validate(schema, data);
+  if (!valid) {
+    const message = `${schemaName} validation failed: ${ajv.errorsText()}`;
+    console.log(message);
+    console.log(JSON.stringify(data, undefined, 2));
+    t.fail(message);
+    throw new Error(message);
+  }
+}
+
+/**
+ * Validate a task input object using json-schema
+ *
+ * Issues a test failure if there were validation errors
+ *
+ * @param {Object} t - an ava test
+ * @param {Object} data - the object to be validated
+ * @returns {Promise<boolean>} - whether the object is valid or not
```

Comment:
So it returns Promise<undefined> now, right? Same with validateConfig and validateOutput

Reply:
Good catch, fixed in 79ffebaaf